### PR TITLE
Start to configure the prompt

### PR DESCRIPTION
**Pull request by alpian**
Made the `$LEDGER` and `$REGION` settable on the prompt, which can only be read from Config for now. I think the ownership model is wrong, but not sure how to resolve it yet. Submitting to be educated! :)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## [2.0.0-alpha6] - [Unreleased]
 
+ * The prompt can now be configured in the config file with `$LEDGER` and `$REGION`. e.g: `prompt="$REGION/$LEDGER > "`
```
Comment (marcbowes) [inline CHANGELOG.md:5]: Please move this under an `## Added` section. My bad for not pre-creating it.

File: src/runner.rs [hunk 2]
```diff
@@ -71,13 +86,7 @@ When your transaction is complete, enter 'commit' or 'abort' as appropriate."#,
...
     #[instrument]
     pub(crate) async fn tick(&mut self) -> Result<bool> {
-        match self.current_transaction {
-            None => self
-                .deps
-                .ui
-                .set_prompt(format!("{} ", self.deps.env.prompt.value)),
-            Some(_) => self.deps.ui.set_prompt(format!("qldb *> ")),
-        }
+    self.deps.ui.set_prompt(set_prompt(&self.deps.env, self.current_transaction.is_some()));
```
Comment (marcbowes) [inline src/runner.rs:89]: spacing seems off

File: src/rusoto_driver.rs [hunk 3]
```diff
@@ -118,7 +118,11 @@ fn rusoto_region(env: &Environment) -> Result<Region> {
             name: Region::default().name().to_owned(),
             endpoint: e.to_owned(),
         },
-        (None, None) => Region::default(),
+        (None, None) => {
+            let region = Region::default();
+            env.set_region(region.name().to_string(), Setter::Environment);
+            region
+        }
     };
```
Comment (marcbowes) [inline src/rusoto_driver.rs:126]: I think this method should just return the region, and runner should then set it on the env. That will remove all the mut stuff.

File: src/rusoto_driver.rs [hunk 4]
```diff
@@ -49,9 +49,9 @@ pub(crate) async fn health_check_start_session(env: &Environment) -> Result<()>
     Ok(())
 }
 
-async fn build_rusoto_client(env: &Environment) -> Result<QldbSessionClient> {
+async fn build_rusoto_client(env: &mut Environment) -> Result<QldbSessionClient> {
     let provider = profile_provider(&env)?;
-    let region = rusoto_region(&env)?;
+    let region = rusoto_region(env)?;
```
Comment (marcbowes) [inline src/rusoto_driver.rs:54]: We should pass the region in as a param.

**Comment by alpian** [comment]
Later changes have made this PR obsolete.

**Pull request closed without merging**